### DISCUSSION TITLE
Clarify how to handle web socket back pressure

### DISCUSSION THREAD

**Discussion by domoritz** [cirospaciari/socketify.py#160] [Q&A] [answered]
The docs should be clearer about how to handle back pressure. Right now, the explanation is 

```py
def ws_message(ws, message, opcode):
    #Ok is false if backpressure was built up, wait for drain
    ok = ws.send(message, opcode)
```

But how should I wait for drain? Should I busy wait before returning from the call or wait in the next call or something else?

**Comment by cirospaciari** [comment]
You should queue this in some way and use the `ws_drain` handler to try send it again

**Reply by cirospaciari** [reply, marked as answer]
and to be clear in the future the user should not have to handle this manually is just bad DX

**Reply by cirospaciari** [reply]
this is being remove in the new engine, we will buffer and handle it for you
